dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/export/trackedentity/DefaultTrackedEntityChangeLogService.java

```diff
@@ -103,19 +102,17 @@ public Page<TrackedEntityChangeLog> getTrackedEntityChangeLog(
       @Nullable UID programUid,
       @Nonnull TrackedEntityChangeLogOperationParams operationParams,
       @Nonnull PageParams pageParams)
-      throws NotFoundException, ForbiddenException, BadRequestException {
-    TrackedEntity trackedEntity = trackedEntityService.getTrackedEntity(trackedEntityUid);
-    if (trackedEntity == null) {
-      throw new NotFoundException(TrackedEntity.class, trackedEntityUid.getValue());
-    }
+      throws NotFoundException, ForbiddenException {
+    TrackedEntity trackedEntity =
+        trackedEntityService.getNewTrackedEntity(
+            trackedEntityUid, programUid, TrackedEntityParams.FALSE.withIncludeAttributes(true));
 
     UserDetails currentUser = CurrentUserUtil.getCurrentUserDetails();
     Set<UID> trackedEntityAttributes = Collections.emptySet();
     if (programUid != null) {
       Program program = validateProgram(programUid.getValue());
       validateOwnership(currentUser, trackedEntity, program);
     } else {
-      validateTrackedEntity(currentUser, trackedEntity);
       trackedEntityAttributes = validateTrackedEntityAttributes(trackedEntity);
```

Comment:
is that still needed as ACL on TEA might not otherwise be done?

Reply:
Maybe @muilpp can help us understand this better.
This validation is just checking if there are teas, and, if not, it is returning a tracked entity not found.

Reply:
@muilpp I was asking about `validateTrackedEntityAttributes`.
The other validation is not needed anymore as it is now present in `TrackerEntityService`